File: src/main/julia/day02.jl
include("intcomp.jl")

prog = map(x->parse(Int64, x), split(readlines("input02.txt")[1], ","))
c = IntComputer(prog)

function part1()
    c.mem[2:3] = [12,02]
    run!(c)
    c.mem[1]
end

function part2() 
    for p in [(noun,verb) for noun=0:99 for verb=0:99]
        c.mem = copy(prog)
        c.mem[2:3] = [p...]
        c.iptr = 1

        run!(c)
        if (c.mem[1] == 19690720)
            return p
        end
    end
end

part1()
part2()

File: src/main/python/2019/day02.py

from intcomp import IntComputer

with open("input02.txt") as f:
    data = [int(x) for x in f.readlines()[0].split(",")] 

# c = IntComputer([1,1,1,4,99,5,6,0,99])

c = IntComputer(data)
c.mem[1:3] = 12, 2
c.run()
print(c.mem[0:10])

# === part 2

for noun in range(100):
    for verb in range(100):
        c = IntComputer(data)
        c.mem[1:3] = noun, verb
        c.run()
        if c.mem[0] == 19690720: print(c.mem[1:3])
